monitoring/api/labeled_custom_metric.py

```diff
@@ -0,0 +1,198 @@
+        }
+    }
+    # Specify a new data point for the time series.
+    timeseries_data = {
+        "timeseriesDesc": timeseries_descriptor,
+        "point": {
+            "start": now_rfc3339,
+            "end": now_rfc3339,
+            "int64Value": count,
+        }
+    }
+    # Submit the write request.
+    request = client.timeseries().write(
+        project=project_id, body={"timeseries": [timeseries_data, ]})
+    try:
+        request.execute()  # ignore the response
+    except Exception as e:
+        print("Failed to write data to custom metric: exception={}".format(e))
+        raise  # propagate exception
+
+
+def read_custom_metric(client, project_id, now_rfc3339, color, size):
+    """Read all the timeseries data points for a given set of label values."""
+    # To identify a time series, specify values for in label as a list.
+    labels = ["%s/color==%s" % (CUSTOM_METRIC_DOMAIN, color),
+              "%s/size==%s" % (CUSTOM_METRIC_DOMAIN, size), ]
+    # Submit the read request.
+    request = client.timeseries().list(
+        project=project_id,
+        metric=CUSTOM_METRIC_NAME,
+        youngest=now_rfc3339,
+        labels=labels)
+    # When a custom metric is created, it may take a few seconds
+    # to propagate throughout the system. Retry a few times.
+    start = time.time()
+    while True:
+        try:
+            response = request.execute()
+            for point in response["timeseries"][0]["points"]:
+                print ("  %s: %s" % (point["end"], point["int64Value"]))
+            break
+        except Exception as e:
+            if time.time() < start + 20:
+                print("Failed to read custom metric data, retrying...")
+                time.sleep(3)
+            else:
+                print("Failed to read custom metric data, aborting: "
+                      "exception={}".format(e))
+                raise  # propagate exception
+
+
+def get_client():
+    """Builds an http client authenticated with the service account
+    credentials."""
+    credentials = GoogleCredentials.get_application_default()
+    client = discovery.build(
+        'cloudmonitoring', 'v2beta2',
+        credentials=credentials)
+    return client
+
+
+def main(project_id, color, size, count):
+    # Assign some values that will be used repeatedly.
+    now_rfc3339 = get_now_rfc3339()
+
+    client = get_client()
+
+    try:
+        print ("Labels: color: %s, size: %s." % (color, size))
+        print ("Creating custom metric...")
+        create_custom_metric(client, project_id)
+        time.sleep(2)
+        print ("Writing new data to custom metric timeseries...")
+        write_custom_metric(client, project_id, now_rfc3339,
+                            color, size, count)
+        print ("Reading data from custom metric timeseries...")
+        write_custom_metric(client, project_id, now_rfc3339, color,
+                            size, count)
+    except Exception as e:
```

Comment:
If you're just going to print the exception, you might as well throw away the `try..catch`.